modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/OnflightMessageTracker.java

```diff
@@ -878,4 +906,70 @@ public void dumpMessageStatusToFile(File fileToWrite) throws AndesException {
         }
     }
 
+    /**
+     * Concurrent List like implementation.
+     * This implementation will create thread safe LinkedList.
+     * Two threads won't be able to execute same operation simultaneously
+     * with this LinkedList.
+     *
+     * @param <T> Type of list items
+     *
+     */
+    private static class ConcurrentTrackingList<T> {
```

Comment:
Why not use http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#synchronizedList(java.util.List) instead of this ? any special reasons? 
@Asitha @a5anka